social_core/backends/base.py

```diff
@@ -148,12 +148,24 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):
     def auth_allowed(self, response, details):
         """Return True if the user should be allowed to authenticate, by
         default check if email is whitelisted (if there's a whitelist)"""
-        emails = self.setting('WHITELISTED_EMAILS', [])
-        domains = self.setting('WHITELISTED_DOMAINS', [])
-        email = details.get('email')
+        def lower_email_domain(email):
+            """Lowercase an email safely by leaving the local-part alone and
+            only lowercasing the domain."""
+            if email is None:
+                return
+
+            try:
+                local_part, domain = email.split('@', 1)
+            except ValueError:
+                return email  # the email is malformed, pass it through unchanged
+            return '@'.join([local_part, domain.lower()])
```

Comment:
I'd assume that whole email is case-insentivite. That would make the whole code a lot simpler and matches reality, see https://stackoverflow.com/a/9808332/225718